### PR TITLE
test(web): add unit tests for Priority 1 pure utilities

### LINKED ISSUE #1929: frontend: add test coverage for Priority 1 utility functions
## Position in sequence

Immediate next step after #1927 — these are pure functions in `lib/` with no entanglement to the god-component refactor track (separate decomposition issue, to be filed). Highest-ROI testing work in the entire frontend coverage initiative. Can ship in parallel with the decomposition work and #1930.

### Moved to #1931

The following files were originally listed here but contain React hooks that need `renderHook` + mocking — they belong in the hook-coverage issue:
- `lib/speedUnits.ts` (the `useSpeedUnits` hook lives alongside the pure `formatSpeedWithUnit`; testing them together avoids splitting one source file across two issues)
- `lib/cross-seed-utils.ts` (the `useLocalCrossSeedMatches` hook lives alongside the pure path helpers)
- `hooks/useTorrentExporter.ts` (entirely a hook; pure helpers inside are not exported)

Note on `dateTimeUtils.ts`: its `getStoredPreferences()` helper reads localStorage. The exported format functions (`formatTimestamp`, `formatRelativeTime`, etc.) all accept preferences as a parameter, so they can be tested purely by passing prefs explicitly. `getStoredPreferences()` itself can be tested with a localStorage stub or skipped.

## Problem

Follow-up to #1927. With test infrastructure landed, add coverage for pure utility functions where bugs have the highest blast radius and tests require no mocking.

## Scope

Ships as **a single PR** covering all six files. These are pure test additions with zero business-logic changes — review burden scales with assertion volume, not with PR count, so splitting buys nothing.

| File | Lines | Priority | What to cover |
|------|-------|----------|---------------|
| `lib/column-filter-utils.ts` | 606 | **Critical** | UI filter → qBittorrent expression conversion, unit handling, state expansion, string matching |
| `lib/torrent-utils.ts` | 201 | High | Hash resolution (v1/v2/legacy), tag intersection, size calc |
| `lib/dateTimeUtils.ts` | 324 | High | Timestamp formatting via injected preferences arg; timezone handling; relative time |
| `lib/workflow-utils.ts` | 210 | Medium | Import/export round-trip, unique name generation, JSON validation |
| `lib/tag-editor.ts` | 114 | Medium | Tag state (on/off/mixed), add/remove diff |
| `lib/search-derived-params.ts` | 78 | Medium | Torznab category → search-type mapping |

## Acceptance

- Each file above has a colocated `*.test.ts` covering its exported pure surface
- Edge cases that have caused real bugs (e.g. #1925-class operator slips, unit conversions) have explicit cases
- `make test-frontend` runs and passes

### PR DESCRIPTION
**Pull request by nitrobass24**
## Summary

Closes #1929. Adds vitest coverage for the six pure utility files identified as Priority 1 in the issue. Builds on the test framework from #1936.

| File | Tests | Notes |
|---|---|---|
| `lib/column-filter-utils.ts` | 120 | Expression building, unit conversion, state expansion (#1925 neighborhood) |
| `lib/torrent-utils.ts` | 47 | Hash resolution (v1/v2/legacy/hybrid), tag intersection, common-category/path |
| `lib/dateTimeUtils.ts` | 39 | All eight formatters with explicit prefs + localStorage fallback group |
| `lib/workflow-utils.ts` | 29 | Import/export round-trip, unique-name generator, JSON validation |
| `lib/tag-editor.ts` | 22 | Tri-state checkbox + add/remove diff |
| `lib/search-derived-params.ts` | 26 | Torznab category mapping |
| **Total** | **283** | |

Every `describe` block has an `// Intent:` comment naming the real-world scenario the test protects against, so future readers can tell intent-pinning tests from those that merely mirror the implementation. Three coverage gaps identified during review were filled in `column-filter-utils` (ISO date with TZ offset, empty-string value, unrecognized freeleech values).

Also introduces `web/src/test/mockTorrent.ts` (a `makeTorrent(overrides)` factory) for reuse by later test files.

## What's NOT in this PR

Three files originally listed in #1929 were moved to #1931 because they're React hooks or contain hooks colocated with their pure helpers:
- `lib/speedUnits.ts` (the `useSpeedUnits` hook)
- `lib/cross-seed-utils.ts` (the `useLocalCrossSeedMatches` hook)
- `hooks/useTorrentExporter.ts` (entirely a hook)

#1929's body has been updated to reflect the moved scope.

## Test plan

- [x] `cd web && pnpm test` — 283/283 pass (plus the smoke test from #1936)
- [x] `cd web && pnpm lint` — 0 errors, 146 warnings (matches develop baseline; no new warnings)
- [x] `cd web && pnpm build` — clean production build
- [x] `npx tsc -b` — clean across all project references
- [x] Verified no test files leak into `web/dist/` (production bundle has no `vitest`/`describe(`/test file paths)
- [x] CI green on this PR

## Summary by CodeRabbit

* **Tests**
  * Added suites for column filtering (unit conversion, expression formatting) and date/time formatting (Timezone/locale/relative)
  * Added tests for search type classification, tag management (sorting, tri-state), torrent utilities (hashes, tags, aggregation)
  * Added workflow automation import/export/validation tests and a torrent factory test helper
  * Removed placeholder smoke test

[Review Change Stack](https://app.coderabbit.ai/change-stack/autobrr/qui/pull/1940?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: web/src/lib/__tests__/torrent-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+  getCommonSavePath,
+  getCommonTags,
+  getTorrentDisplayHash,
+  getTorrentHashesWithTag,
+  getTotalSize,
+  normalizeTorrentHash,
+  parseTorrentTags,
+  resolveTorrentHashes,
+  torrentHasTag
+} from "@/lib/torrent-utils"
```
[the remark this answers is not in the document]
Comment (nitrobass24) [reply web/src/lib/__tests__/torrent-utils.test.ts:18]: eslint autofix removes the trailing commas.

File: web/src/lib/__tests__/workflow-utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+  it("includes notify only when it's an explicit boolean", () => {
+    const withNotify = parseImportJSON(JSON.stringify({
+      name: "x",
+      conditions: { schemaVersion: "1" },
+      trackerDomains: [],
+      notify: false,
+    }))
+    expect(withNotify.data?.notify).toBe(false)
+  })
```
[the remark this answers is not in the document]
Comment (nitrobass24) [reply web/src/lib/__tests__/workflow-utils.test.ts:249]: Verified, but this asymmetry might be intentional.

**Review by s0up4200 — APPROVED** [review]
